### PR TITLE
Add a method of enabling or disabling containers using labels

### PR DESCRIPTION
**Pull request by belak**
This would potentially be a better way of handling #102, #95, #58, and #4.

I haven't tested this yet and there may be a cleaner way of doing this, but it seems like a relatively simple concept.

### CODE REVIEW

**Comment by belak** [comment]
This adds two features:

* A label (`com.centurylinklabs.watchtower.enable`) which lets users specify if a container should be enabled or not on a per-container basis.
* A command line option, `--no-enable` which will let watchtower start up without watching any specific containers (unless they're specifically enabled with labels)

**Comment by voglster** [comment]
+1 this, was going to write my own adding functionality similar to this until I saw this PR.  for now I'm testing belak's branch.

File: main.go [hunk 1]
```diff
@@ -114,7 +119,7 @@ func before(c *cli.Context) error {
 		return err
 	}
 
-	client = container.NewClient(!c.GlobalBool("no-pull"))
+	client = container.NewClient(!c.GlobalBool("no-pull"), !c.GlobalBool("no-watch"))
```
Comment (chriswue) [inline main.go]: The command line option is "no-enable" while this looks for "no-watch". In fact I'd probably prefer a slightly more verbose option name like "no-watch-by-default"

**Review by chriswue — CHANGES_REQUESTED** [review]
There is an option name bug which needs fixing but otherwise the changes look good to me.

**Comment by Nightbr** [comment]
Any news on this, it's a real killer feature needed for watchtower!

Add to that, I think we should rename the flag as `--label-enable` in order to only take labeled containers. It makes more sense for me. What do you think?

**Comment by stffabi** [comment]
@belak thanks for your contribution and thanks @chriswue for reviewing it.

I like the idea of @Nightbr to rename the flag to `--label-enable` and then only take labeled containers. I think this would be much more intuitive for the user.

On the other hand having a possibility to exclude is also handy. Maybe we could implement an include and exclude option for named lables, e.g. `--include-label mylabel` and `--exclude-label myExcludeLabel`. Specifying no include labels means all containers, and one can exclude some others based on exclude labels.

And we should also include the new command line parameter and the documentation to the readme.md

**Comment by stffabi** [comment]
Any opinions how we should proceed? Should we go the `include` `exclude` way, or do you prefer some other handling?

**Comment by belak** [comment]
Sorry, I've been busy with other stuff lately. At least for me, the simplest way to handle this would be `--label-enable` and that would be good enough for me. A full include/exclude seems like it might be overcomplicating this.

Also, do you have any instructions on how to run this manually for testing? Do you just use it outside a docker container manually?

File: main.go [hunk 2]
```diff
@@ -79,39 +79,44 @@ func main() {
 			Usage:  "use TLS and verify the remote",
 			EnvVar: "DOCKER_TLS_VERIFY",
 		},
+		cli.BoolFlag{
+			Name:   "label-enable",
+			Usage:  "decide which containers to watch based on a label",
```
Comment (stffabi) [inline main.go]: Could you add which label we use here?

**Review by stffabi — CHANGES_REQUESTED** [review]

**Comment by stffabi** [comment]
@belak I'm fine with that 😄. Would it be possible for you to add the corresponding documentation in the readme.md? And I've put in another minor comment.

Apart from that LGTM, and I'm looking forward to merge this.

**Comment by stffabi** [comment]
For testing I always use it outside of a docker container. That works quite well.

**Comment by belak** [comment]
I'm open to any updates in wording... this was just thrown together based on the other documentation.

**Comment by stffabi** [comment]
LGTM, thanks again for your contribution. I'm going to squash the commits and will then merge it in.

**Review by stffabi — APPROVED** [review]

**Pull request merged**